diff --git a/Modules/_tkinter.c b/Modules/_tkinter.c
index 005036d3ff2..f58fd5b6764 100644
--- a/Modules/_tkinter.c
+++ b/Modules/_tkinter.c
@@ -58,6 +58,9 @@ Copyright (C) 1994 Steen Lumholt.
 #error "Tk older than 8.5.12 not supported"
 #endif
 
+#ifndef TCL_WITH_EXTERNAL_TOMMATH
+#define TCL_NO_TOMMATH_H
+#endif
 #include <tclTomMath.h>
 
 #if defined(TCL_WITH_EXTERNAL_TOMMATH) || (TK_HEX_VERSION >= 0x08070000)
