diff --git a/configure.ac b/configure.ac
index 3fcb18922c5..3be45c99c41 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2260,6 +2260,7 @@ then
      -reorder-blocks=ext-tsp
      -reorder-functions=cdsort
      -split-functions
+     -split-strategy=cdsplit
      -icf=1
      -inline-all
      -split-eh
@@ -2268,9 +2269,11 @@ then
      -jump-tables=aggressive
      -inline-ap
      -indirect-call-promotion=all
+     -plt=all
      -dyno-stats
      -use-gnu-stack
      -frame-opt=hot
+     -hugify
    ")]
   )
 fi
