diff --git a/configure.ac b/configure.ac
index c5dc9bf174a..11631a978ce 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2142,11 +2142,7 @@ if test "$Py_BOLT" = 'true' ; then
   # These flags are required for bolt to work:
   LDFLAGS_NODIST="$LDFLAGS_NODIST -Wl,--emit-relocs"
 
-  # These flags are required to get good performance from bolt:
-  CFLAGS_NODIST="$CFLAGS_NODIST -fno-pie"
-  # We want to add these no-pie flags to linking executables but not shared libraries:
-  LINKCC="$LINKCC -fno-pie -no-pie"
   AC_SUBST([LLVM_BOLT])
   AC_PATH_TOOL([LLVM_BOLT], [llvm-bolt], [''], [${llvm_path}])
   if test -n "${LLVM_BOLT}" -a -x "${LLVM_BOLT}"
