diff --git a/configure.ac b/configure.ac
index ac3be3850a9..88a2027f517 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2713,8 +2713,16 @@ then
 		fi
 		;;
 	Linux*|GNU*|QNX*|VxWorks*)
-		LDSHARED='$(CC) -shared'
-		LDCXXSHARED='$(CXX) -shared';;
+		# See https://maskray.me/blog/2021-05-16-elf-interposition-and-bsymbolic.
+		# This is related to the -fno-semantic-interposition optimization that
+		# --enable-optimizations engages automatically. It changes the linkage
+		# behavior so the runtime loader searches the current DSO first before
+		# falling back to the default search mechanism of the executable plus
+		# all its loaded DSOs. It has much the same effect as
+		# -fno-semantic-interposition but can also enable optimizations across
+		# translation units.
+		LDSHARED='$(CC) -shared -Wl,-Bsymbolic-functions'
+		LDCXXSHARED='$(CXX) -shared -Wl,-Bsymbolic-functions';;
 	FreeBSD*)
 		if [[ "`$CC -dM -E - </dev/null | grep __ELF__`" != "" ]]
 		then
